src/app/routes/_analytics.router.js

```diff
@@ -0,0 +1,210 @@
+        .find({
+          suspendedUntil: { $gt: Date.now() }
+        }, 'name gravatarHash _id country')
+        .lean()
+        .exec();
+
+      const inactive = await User
+        .find({
+          // older than 3 months (90 days) 7776E6
+          lastActive: { $lte: Date.now() - 7776E6 }
+        })
+        .count()
+        .exec();
+
+      res.json({
+        count: count,
+        admins: admins,
+        suspended: suspended,
+        inactive: inactive
+      });
+    } catch (err) {
+      res.status(500).send(err);
+    }
+  });
+
+  /**
+   * @api {get} /api/analytics-countries Get statistics of the countries of all accounts
+   * @apiName GetCountryStats
+   * @apiGroup analytics
+   *
+   * @apiSuccess {Object[]} countries  List of all countries and their count of users
+   */
+  router.get('/api/analytics-countries', admin, async (req, res) => {
+    try {
+      User
+        .aggregate([
+        {
+          $group: {
+            _id: '$country', count: { $sum: 1 }
+          }
+        },
+        { $sort : { count : -1} },
+        { $limit : 10 }
+      ], (err, countries) => {
+        if (err) return handleError(err);
+        res.json(countries);
+      });
+    } catch (err) {
+      console.log(err)
+      res.status(500).send(err);
+    }
+  });
+
+  /**
+   * @api {get} /api/analytics-timezones Get statistics of the timezone of all accounts
+   * @apiName GetTimezoneStats
+   * @apiGroup analytics
+   *
+   * @apiSuccess {Object[]} timezones  List of all timezones and their count of users
+   */
+  router.get('/api/analytics-timezones', admin, async (req, res) => {
+    try {
+      User.aggregate([
+        {
+          $group: {
+            _id: '$timezone', count: { $sum: 1 }
+          }
+        },
+        { $sort : { count : -1} },
+        { $limit : 10 }
+      ], (err, timezones) => {
+        if (err) return handleError(err);
+        res.json(timezones);
+      });
+    } catch (err) {
+      res.status(500).send(err);
+    }
+  });
+
+  /**
+   * @api {get} /api/analytics-signups Get statistics about new accounts
+   * @apiName GetSignupStats
+   * @apiGroup analytics
+   *
+   * @apiSuccess {Object} signupChartData  Data for linechart
+   */
+  router.post('/api/analytics-signups', admin, async (req, res) => {
+    try {
+      // Default interval: 1 day (= 864E5 ms)
+      const interval = req.body.interval ? req.body.interval : 864E5;
+
+      // Default interval: Today - 7 days (= 6048E5 ms)
+      const minDate = req.body.minDate ? req.body.minDate : Date.now() - 6048E5;
+      const dtFormat = req.body.format ? req.body.format : 'MM/DD/YY';
+
+      let data = {
+        data: [],
+        labels: []
+      };
+
+      let dtEnd = Date.now();
+      let dtStart = dtEnd - interval;
+
+      while (dtStart > minDate) {
+        // Use MongoDB's id to find new users within the current timespan
+        const minObjID = ObjectId(Math.floor(dtStart / 1000).toString(16) + "0000000000000000");
```

Comment:
http://stackoverflow.com/questions/6452021/getting-timestamp-from-mongodb-id The first 4 Bytes of a MongoDB id contain a timestamp of the creation. That means when an object id is greater than another one is, its creation datetime of the first one has to be after (or at the same time) the creation of the second one.